### PR TITLE
QE: Add an error detection on the product catalog page

### PR DESCRIPTION
**Pull request by elariekerboull**
## What does this PR change?

Adds a scenario detecting warning signs on the UI when an error occurs in the product catalog page
Related to https://github.com/SUSE/spacewalk/issues/16987

## GUI diff
No difference.
- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes
- [x] **DONE**

## Test coverage
- Cucumber tests were added
- [x] **DONE**

## Links

Fixes # https://github.com/SUSE/spacewalk/issues/16987
Tracks # 
4.2 https://github.com/SUSE/spacewalk/pull/20186
4.3 https://github.com/SUSE/spacewalk/pull/20185
- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [X] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: testsuite/features/build_validation/reposync/srv_sync_all_products.feature [hunk 1]
```diff
@@ -233,3 +233,8 @@ Scenario: Add RHEL and Liberty 9
     When I click the Add Product button
     And I wait until I see "Selected channels/products were scheduled successfully for syncing." text
     And I wait until I see "SUSE Manager Retail Branch Server 4.3 x86_64" product has been added
+
+  Scenario: Detect product loading issues from the UI in Build Validation
+    When I follow the left menu "Admin > Setup Wizard > Products"
+    Then I should not see a "Operation not successful" text
+    And I should not see a warning sign
```
Comment (Bischoff) [inline testsuite/features/build_validation/reposync/srv_sync_all_products.feature]: missing linefeed on last line

File: testsuite/features/reposync/srv_sync_products.feature [hunk 2]
```diff
@@ -78,3 +78,9 @@ Feature: Synchronize products in the products page of the Setup Wizard
     When I execute mgr-sync "list channels" with user "admin" and password "admin"
     Then I should get "    [I] SLES12-SP5-Installer-Updates for x86_64 SUSE Linux Enterprise Server 12 SP5 x86_64 [sles12-sp5-installer-updates-x86_64]"
     And I should get "    [I] SLE15-SP4-Installer-Updates for x86_64 SUSE Linux Enterprise Server 15 SP4 x86_64 [sle15-sp4-installer-updates-x86_64]"
+
+@scc_credentials
+  Scenario: Detect product loading issues from the UI
+    When I follow the left menu "Admin > Setup Wizard > Products"
+    Then I should not see a "Operation not successful" text
+    And I should not see a warning sign
```
Comment (Bischoff) [inline testsuite/features/reposync/srv_sync_products.feature]: missing final linefeed

File: testsuite/features/step_definitions/navigation_steps.rb [hunk 3]
```diff
@@ -748,6 +748,10 @@
   end
 end
 
+Then(/^I should not see a warning sign$/) do
+  raise 'Errors detected in the Products Catalog' unless page.has_no_xpath?("//*[contains(@class, 'fa fa-li fa-exclamation-triangle text-warning')]", wait: 5)
+end
+
```
Comment (Bischoff) [inline testsuite/features/step_definitions/navigation_steps.rb:754]: Cool xpath usage

Why this timeout ? Maybe we could use the default timeout for capybara? It's defined as:
```
Capybara.default_max_wait_time = ENV['CAPYBARA_TIMEOUT'] ? ENV['CAPYBARA_TIMEOUT'].to_i : 10
```

File: testsuite/features/reposync/srv_sync_products.feature (hunk 2 above)
Comment (srbarrios) [inline testsuite/features/reposync/srv_sync_products.feature:86]: It seems it needs an additional new line.

File: testsuite/features/step_definitions/navigation_steps.rb [hunk 4]
```diff
@@ -748,6 +748,10 @@
   end
 end
 
+Then(/^I should not see a warning sign$/) do
+  raise 'Errors detected in the Products Catalog' unless page.has_no_xpath?("//*[contains(@class, 'fa fa-li fa-exclamation-triangle text-warning')]", wait: 5)
```
Comment (srbarrios) [inline testsuite/features/step_definitions/navigation_steps.rb:752]:
```suggestion
  raise 'Warning detected' unless page.has_no_xpath?("//*[contains(@class, 'fa fa-li fa-exclamation-triangle text-warning')]", wait: 5)
```

We could be using this step in other scenarios where we have warning signs. Let's be generic in the error description.

**Review by srbarrios — APPROVED** [review]
LGTM. Once my suggestions are addressed.

**Review by Bischoff — DISMISSED** [review]
see comments

File: testsuite/features/step_definitions/navigation_steps.rb (hunk 3 above)
Comment (elariekerboull) [reply testsuite/features/step_definitions/navigation_steps.rb:754]: Agreed, removed the timeout and will look into applying this to some other similar definedsteps

**Pull request merged**